### DISCUSSION TITLE
Docs: Upgrade guide for CodeClimate users

### DISCUSSION THREAD

**Discussion by ukch** [qltysh/qlty#1303] [Feature Requests]
I have been using CodeClimate for a few years, mostly to track test coverage, but also for the quality tools. I read the announcement about qlty and I see the new CLI and cloud platform are now live, but I don't see any upgrade guide for those of us still using CodeClimate. 

Please can this be provided so we can enjoy the new product?

**Comment by brynary** [comment]
Thank you for this suggestion, @ukch. Yes, we will add an upgrade guide for moving from Code Climate Quality to Qlty this week.

Today we have [documentation](https://docs.qlty.sh) but we are missing the upgrade guide. In the meantime, here are a few pointers which may be helpful...

We built Qlty to be able to run side-by-side with Code Climate Quality so that you can try it out before making the switch. When you log in to Qlty, your projects should be there waiting for you. (Soon we will make it easier to switch back and forth.)

Qlty has a new, updated set of static analysis plugins and uses a different configuration file (`.qlty/qlty.toml`). We have some limited support for transforming a `.codeclimate.yml` configuration into a Qlty configuration which you can try out like this:

```sh
# Install Qlty CLI
curl https://qlty.sh | sh

# Generate a qlty.toml based on repository contents 
qlty init

# Apply some settings from an existing .codeclimate.yml into the  qlty.toml
qlty config migrate

# Install linters (and runtimes to run them)
qlty install

# Run linters, sample the results
qlty check --sample 5

# Auto-format all files
qlty fmt --all
```

If your project has a qlty.toml checked in, we use that when analyzing it on the cloud. Otherwise we generate one each time.

In the Project Settings area in Qlty Cloud you can turn on GitHuh integration (PR analysis, commit statuses and code review comments). We also have some early access features we can turn on for your account if you tell us the GitHub organization name.

Code coverage requires additional setup in the CI scripts, unfortunately. You can get your code coverage upload token from within Qlty Cloud.

Please let us know if you have any questions or run into any issues and we would be happy to help!

**Comment by tsaavik** [comment]
`qlty smells` [seems to be](https://docs.qlty.sh/cli/commands/smells) the replacement for `codeclimate analyze`

**Comment by brynary** [comment]
Update: We now have a [Migration Guide](https://docs.qlty.sh/migration/guide) available.

Also, we have a [Coverage Migraiton Guide](https://docs.qlty.sh/migration/coverage) for updating CI workflows to send coverage data to Qlty.
